utoipa/src/openapi/path.rs

```diff
@@ -627,7 +637,81 @@ pub enum ParameterStyle {
 #[cfg(test)]
 mod tests {
     use super::{Operation, OperationBuilder};
-    use crate::openapi::{security::SecurityRequirement, server::Server};
+    use crate::openapi::{
+        security::SecurityRequirement, server::Server, PathItem, PathItemType, PathsBuilder,
+    };
+
+    #[test]
+    fn test_path_order() {
+        let paths_list = PathsBuilder::new()
+            .path(
+                "/todo",
+                PathItem::new(PathItemType::Get, OperationBuilder::new()),
+            )
+            .path(
+                "/todo",
+                PathItem::new(PathItemType::Post, OperationBuilder::new()),
+            )
+            .path(
+                "/todo/{id}",
+                PathItem::new(PathItemType::Delete, OperationBuilder::new()),
+            )
+            .path(
+                "/todo/{id}",
+                PathItem::new(PathItemType::Get, OperationBuilder::new()),
+            )
+            .path(
+                "/todo/{id}",
+                PathItem::new(PathItemType::Put, OperationBuilder::new()),
+            )
+            .path(
+                "/todo/search",
+                PathItem::new(PathItemType::Get, OperationBuilder::new()),
+            )
+            .build();
+
+        let mut actual_value = Vec::new();
+
+        for path in paths_list.paths.keys() {
+            let path_item = paths_list.paths.get(path);
+            if let Some(path_item) = path_item {
+                for method in path_item.operations.keys() {
+                    actual_value.push((path.as_str(), method));
+                }
+            }
+        }
```

Comment:
Nested loops are not fun and can be hard to read and follow the code progress. I'd recommed to use the Rust's iterator API as much as possible as it provides fantastic functional interface to operate on collections of data and even more. 
```suggestion
        let actual_value = paths_list
            .paths
            .iter()
            .flat_map(|(path, path_item)| {
                path_item.operations.iter().fold(
                    Vec::<(&str, &PathItemType)>::with_capacity(paths_list.paths.len()),
                    |mut acc, (method, _)| {
                        acc.push((path.as_str(), method));
                        acc
                    },
                )
            })
            .collect::<Vec<_>>();
```

Reply:
I'm new to Rust programming and am still in the learning phase. I appreciate your inputs. I'll make the necessary changes and thanks for your suggestion.

Reply:
That's fine, :slightly_smiling_face: we all have been there at some point not matter the language. The willingness to learn and constantly improve is the key :+1: